target/linux/ath79/dts/ar9330_teltonika_rut230.dts

```diff
@@ -0,0 +1,204 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+#include "ar9330.dtsi"
+
+/ {
+	compatible = "teltonika,rut230", "qca,ar9330";
+	model = "Teltonika RUT230";
+
+	chosen {
+		bootargs = "console=ttyATH0,115200 root=31:05 rootfstype=squashfs";
+	};
+
+	aliases {
+		serial0 = &uart;
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "reset";
+			linux,code = <KEY_RESTART>;
+			gpios = <&gpio 22 GPIO_ACTIVE_HIGH>;
+			debounce-interval = <60>;
+		};
+
+		input {
+			label = "input";
+			linux,code = <BTN_0>;
+			gpios = <&gpio 16 GPIO_ACTIVE_LOW>;
+			debounce-interval = <60>;
+		};
+
+		sim {
+			label = "SIM holder";
+			linux,code = <KEY_RFKILL>;
+			gpios = <&gpio 20 GPIO_ACTIVE_LOW>;
+			debounce-interval = <60>;
+		};
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		signal1 {
+			label = "rut230:green:signal1";
+			gpios = <&gpio 23 GPIO_ACTIVE_LOW>;
+		};
+
+		signal2 {
+			label = "rut230:green:signal2";
+			gpios = <&gpio 7 GPIO_ACTIVE_LOW>;
+		};
+
+		signal3 {
+			label = "rut230:green:signal3";
+			gpios = <&gpio 6 GPIO_ACTIVE_LOW>;
+		};
+
+		signal4 {
+			label = "rut230:green:signal4";
+			gpios = <&gpio 26 GPIO_ACTIVE_LOW>;
+		};
+
+		signal5 {
+			label = "rut230:green:signal5";
+			gpios = <&gpio 27 GPIO_ACTIVE_LOW>;
+		};
+
+		status_2g {
+			label = "rut230:green:status_2g";
+			gpios = <&gpio 8 GPIO_ACTIVE_HIGH>;
+		};
+
+		status_3g {
+			label = "rut230:green:status_3g";
+			gpios = <&gpio 24 GPIO_ACTIVE_HIGH>;
+		};
+
+		status_4g {
```

Comment:
I don't find this LED in the manual?

Reply:
tatic struct gpio_led tlt_rut200_leds_gpio[] __initdata = {
  1         {
  2                 .name           = "signal_bar0",
  3                 .gpio           = TLT_RUT200_GPIO_LED_SS1,
  4                 .active_low     = 1,
  5         }, {
  6                 .name           = "signal_bar1",
  7                 .gpio           = TLT_RUT200_GPIO_LED_SS2,
  8                 .active_low     = 1,
  9         }, {
 10                 .name           = "signal_bar2",
 11                 .gpio           = TLT_RUT200_GPIO_LED_SS3,
 12                 .active_low     = 1,
 13         }, {
 14                 .name           = "signal_bar3",
 15                 .gpio           = TLT_RUT200_GPIO_LED_SS4,
 16                 .active_low     = 1,
 17         }, {
 18                 .name           = "signal_bar4",
 19                 .gpio           = TLT_RUT200_GPIO_LED_SS5,
 20                 .active_low     = 1,
 21         }, {
 22                 .name           = "status_2g",
 23                 .gpio           = TLT_RUT200_GPIO_LED_2G,
 24                 .active_low     = 0,
 25         }, {
 26                 .name           = "status_3g",
 27                 .gpio           = TLT_RUT200_GPIO_LED_3G,
 28                 .active_low     = 0,
 29         }, {
 30                 .name           = "status_4g",
 31                 .gpio           = TLT_RUT200_GPIO_LED_4G,
 32                 .active_low     = 0,
 33         }, {
 34                 .name           = "led_lan",
 35                 .gpio           = 13,
 36                 .active_low     = 0,
 37         }, {
 38                 .name           = "led_wan",
 39                 .gpio           = 14,
 40                 .active_low     = 0,
 41         },
 42 };
This was hardcoded into vendors firmware

Reply:
But i think it is added for RUT240 device witch has 4g and the extra LED witch is the only difference from RUT230. I can remove it, but this won't be compatible with 240

Reply:
In addition they use same firmware for both devices

Reply:
Yes, I understand that you correctly ported this from their code (as it appears to me).

Still, if the two devices have different features, we should give them different images here. So, I'd limit the coded features for the rut230 to those the rut230 actually has.

Reply:
Sure, I agree. Removed it.